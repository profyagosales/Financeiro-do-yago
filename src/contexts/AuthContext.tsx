import type { User } from '@supabase/supabase-js';
import { createContext, useContext, useEffect, useState } from 'react';

import { supabase } from '@/lib/supabaseClient';

export type AuthContextValue = {
  user: User | null;
  loading: boolean;
  signOut: () => Promise<void>;
  // expostos para o Login.tsx
  loginWithPassword: (email: string, password: string) => Promise<void>;
  signUpWithPassword: (email: string, password: string) => Promise<void>;
};

const AuthContext = createContext<AuthContextValue | undefined>(undefined);

export function AuthProvider({ children }: { children: React.ReactNode }) {
  const [user, setUser] = useState<User | null>(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    let mounted = true;

    supabase.auth.getSession().then(({ data }) => {
      if (!mounted) return;
      setUser(data.session?.user ?? null);
      setLoading(false);
    });

    const {
      data: { subscription },
    } = supabase.auth.onAuthStateChange((_event, session) => {
      setUser(session?.user ?? null);
      setLoading(false);
    });

    return () => {
      mounted = false;
      subscription.unsubscribe();
    };
  }, []);

  const loginWithPassword = async (email: string, password: string) => {
    const { error } = await supabase.auth.signInWithPassword({ email, password });
    if (error) throw error;
  };

  const signUpWithPassword = async (email: string, password: string) => {
    const { error } = await supabase.auth.signUp({
      email,
      password,
      options: { emailRedirectTo: `${window.location.origin}/confirm` },
    });
    if (error) throw error;
  };

  const signOut = async () => {
    await supabase.auth.signOut();
  };

  const value: AuthContextValue = {
    user,
    loading,
    signOut,
    loginWithPassword,
    signUpWithPassword,
  };

  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
}

export function useAuth() {
  const ctx = useContext(AuthContext);
  if (!ctx) throw new Error('useAuth deve ser usado dentro de <AuthProvider>');
  return ctx;
}
