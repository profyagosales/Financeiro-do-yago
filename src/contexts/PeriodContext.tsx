/**
 * src/contexts/PeriodContext.tsx
 * Shim de compatibilidade: reexporta a implementação real do estado.
 * Assim, imports antigos de "@/contexts/PeriodContext" seguem funcionando
 * mesmo que a fonte oficial esteja em "@/state/periodFilter".
 */
export { PeriodProvider, usePeriod } from "@/state/periodFilter";
