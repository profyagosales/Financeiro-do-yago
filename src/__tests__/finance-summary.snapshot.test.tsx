import { render } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { describe, expect, it } from 'vitest';

import FinanceSummary from '@/pages/FinanceSummary';
import { PeriodProvider } from '@/contexts/PeriodContext';

// Snapshot simples para garantir estrutura inicial da página de resumo de finanças.

describe('FinanceSummary snapshot', () => {
  it('matches structure', () => {
    const { asFragment } = render(
      <PeriodProvider>
        <MemoryRouter initialEntries={['/financas/resumo']}>
          <Routes>
            <Route path='/financas/resumo' element={<FinanceSummary />} />
          </Routes>
        </MemoryRouter>
      </PeriodProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
