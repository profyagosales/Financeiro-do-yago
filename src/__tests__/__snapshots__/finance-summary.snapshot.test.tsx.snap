// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FinanceSummary snapshot > matches structure 1`] = `
<DocumentFragment>
  <div
    class="space-y-6"
    data-clr="financas"
    style="background-image: radial-gradient(at 30% 20%, var(--clr-active)/15%, transparent 70%); --section-fg: #0f172a; color: var(--section-fg);"
  >
    <div
      class="mb-6 rounded-2xl text-white backdrop-blur-sm border border-white/10 overflow-hidden relative bg-[linear-gradient(90deg,var(--clr-active)_0%,var(--clr-active)_55%,rgba(255,255,255,0.15)_100%)] dark:bg-[linear-gradient(90deg,var(--clr-active)_0%,var(--clr-active)_50%,rgba(255,255,255,0.08)_100%)]"
      style="opacity: 0; transform: translateY(-10px);"
    >
      <div
        class="container mx-auto px-4 py-6 flex items-center justify-between gap-4"
      >
        <div
          class="flex items-center gap-3 min-w-0"
        >
          <div
            class="min-w-0"
          >
            <h1
              class="text-xl sm:text-2xl font-semibold tracking-tight truncate drop-shadow-sm"
            >
              Finanças — Resumo
            </h1>
            <div
              class="mt-1 flex flex-wrap items-center gap-1 text-xs text-white/80"
            >
              <span
                class="inline-flex items-center gap-1"
              >
                <a
                  class="underline"
                  href="/financas"
                >
                  Finanças
                </a>
              </span>
              <span
                class="inline-flex items-center gap-1"
              >
                <span>
                  /
                </span>
                <span>
                  Resumo
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="shrink-0"
        >
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 h-9 px-4 py-2"
            style="opacity: 0; transform: translateY(-4px);"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-download mr-2 size-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 15V3"
              />
              <path
                d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
              />
              <path
                d="m7 10 5 5 5-5"
              />
            </svg>
             Exportar PDF
          </button>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col gap-2 sm:flex-row sm:items-center"
    />
    <div
      class="grid grid-cols-12 gap-5"
    >
      <div
        class="col-span-12 sm:col-span-6 lg:col-span-4 xl:col-span-2 rounded-xl border border-white/10 bg-background/70 backdrop-blur p-4 flex flex-col justify-center min-h-[110px]"
      >
        <div
          class="flex items-center gap-3"
        >
          <div
            class="rounded-lg bg-primary/90 p-2 text-primary-foreground"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-wallet size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 7V4a1 1 0 0 0-1-1H5a2 2 0 0 0 0 4h15a1 1 0 0 1 1 1v4h-3a2 2 0 0 0 0 4h3a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1"
              />
              <path
                d="M3 5v14a2 2 0 0 0 2 2h15a1 1 0 0 0 1-1v-4"
              />
            </svg>
          </div>
          <div
            class="flex flex-col leading-tight"
          >
            <span
              class="text-xs font-medium uppercase tracking-wide text-muted-foreground"
            >
              Saldo
            </span>
            <div
              class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] mt-1 h-6 w-24"
            />
          </div>
        </div>
      </div>
      <div
        class="col-span-12 sm:col-span-6 lg:col-span-4 xl:col-span-2 rounded-xl border border-white/10 bg-background/70 backdrop-blur p-4 flex flex-col justify-center min-h-[110px]"
      >
        <div
          class="flex items-center gap-3"
        >
          <div
            class="rounded-lg bg-primary/90 p-2 text-primary-foreground"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-circle-arrow-up size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="m16 12-4-4-4 4"
              />
              <path
                d="M12 16V8"
              />
            </svg>
          </div>
          <div
            class="flex flex-col leading-tight"
          >
            <span
              class="text-xs font-medium uppercase tracking-wide text-muted-foreground"
            >
              Entradas
            </span>
            <div
              class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] mt-1 h-6 w-24"
            />
          </div>
        </div>
      </div>
      <div
        class="col-span-12 sm:col-span-6 lg:col-span-4 xl:col-span-2 rounded-xl border border-white/10 bg-background/70 backdrop-blur p-4 flex flex-col justify-center min-h-[110px]"
      >
        <div
          class="flex items-center gap-3"
        >
          <div
            class="rounded-lg bg-primary/90 p-2 text-primary-foreground"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-circle-arrow-down size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M12 8v8"
              />
              <path
                d="m8 12 4 4 4-4"
              />
            </svg>
          </div>
          <div
            class="flex flex-col leading-tight"
          >
            <span
              class="text-xs font-medium uppercase tracking-wide text-muted-foreground"
            >
              Saídas
            </span>
            <div
              class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] mt-1 h-6 w-24"
            />
          </div>
        </div>
      </div>
      <div
        class="col-span-12 sm:col-span-6 lg:col-span-4 xl:col-span-2 rounded-xl border border-white/10 bg-background/70 backdrop-blur p-4 flex flex-col justify-center min-h-[110px]"
      >
        <div
          class="flex items-center gap-3"
        >
          <div
            class="rounded-lg bg-primary/90 p-2 text-primary-foreground"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-receipt size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 2v20l2-1 2 1 2-1 2 1 2-1 2 1 2-1 2 1V2l-2 1-2-1-2 1-2-1-2 1-2-1-2 1Z"
              />
              <path
                d="M16 8h-6a2 2 0 1 0 0 4h4a2 2 0 1 1 0 4H8"
              />
              <path
                d="M12 17.5v-11"
              />
            </svg>
          </div>
          <div
            class="flex flex-col leading-tight"
          >
            <span
              class="text-xs font-medium uppercase tracking-wide text-muted-foreground"
            >
              Ticket médio
            </span>
            <div
              class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] mt-1 h-6 w-24"
            />
          </div>
        </div>
      </div>
      <div
        class="col-span-12 sm:col-span-6 lg:col-span-4 xl:col-span-2 rounded-xl border border-white/10 bg-background/70 backdrop-blur p-4 flex flex-col justify-center min-h-[110px]"
      >
        <div
          class="flex items-center gap-3"
        >
          <div
            class="rounded-lg bg-primary/90 p-2 text-primary-foreground"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-list-todo size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="6"
                rx="1"
                width="6"
                x="3"
                y="5"
              />
              <path
                d="m3 17 2 2 4-4"
              />
              <path
                d="M13 6h8"
              />
              <path
                d="M13 12h8"
              />
              <path
                d="M13 18h8"
              />
            </svg>
          </div>
          <div
            class="flex flex-col leading-tight"
          >
            <span
              class="text-xs font-medium uppercase tracking-wide text-muted-foreground"
            >
              Transações do mês
            </span>
            <div
              class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] mt-1 h-6 w-24"
            />
          </div>
        </div>
      </div>
      <div
        class="col-span-12 sm:col-span-6 lg:col-span-4 xl:col-span-2 rounded-xl border border-white/10 bg-background/70 backdrop-blur p-4 flex flex-col justify-center min-h-[110px]"
      >
        <div
          class="flex items-center gap-3"
        >
          <div
            class="rounded-lg bg-primary/90 p-2 text-primary-foreground"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-calendar-clock mr-2 size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 14v2.2l1.6 1"
              />
              <path
                d="M16 2v4"
              />
              <path
                d="M21 7.5V6a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h3.5"
              />
              <path
                d="M3 10h5"
              />
              <path
                d="M8 2v4"
              />
              <circle
                cx="16"
                cy="16"
                r="6"
              />
            </svg>
          </div>
          <div
            class="flex flex-col leading-tight"
          >
            <span
              class="text-xs font-medium uppercase tracking-wide text-muted-foreground"
            >
              A receber
            </span>
            <div
              class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] mt-1 h-6 w-24"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid gap-6 lg:grid-cols-12"
    >
      <div
        class="rounded-lg bg-white shadow-sm p-6  u-card-base lg:col-span-6"
        style="opacity: 0; transform: translateY(16px);"
      >
        <div
          class="mb-3 flex items-center gap-2"
        >
          <h3
            class="text-lg font-semibold"
          >
            Previsão — 30 dias
          </h3>
          <span
            class="size-4 text-zinc-400"
          />
          Estimativa simples baseada na média dos últimos dias
        </div>
        <div
          class="flex flex-col items-center justify-center gap-2 py-10 text-center text-fg-muted"
        >
          <h3
            class="text-sm font-medium"
          >
            Sem dados
          </h3>
        </div>
      </div>
      <div
        class="rounded-lg bg-white shadow-sm p-6  u-card-base lg:col-span-6"
        style="opacity: 0; transform: translateY(16px);"
      >
        <div
          class="mb-3"
        >
          <h3
            class="text-lg font-semibold"
          >
            Fluxo de caixa mensal
          </h3>
        </div>
        <div
          class="rounded-xl border bg-white dark:bg-slate-900 p-4"
        >
          <h3
            class="font-display tracking-tight text-text dark:text-text-dark text-xl font-semibold mb-3"
          >
            Movimento diário
          </h3>
          <div
            class="h-[320px]"
          >
            <div
              class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-full w-full"
            />
          </div>
        </div>
        <a
          class="mt-3 inline-flex items-center gap-1 text-sm font-medium text-emerald-700 hover:underline"
          href="/financas/mensal"
        >
          Ver detalhes
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right size-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </a>
      </div>
      <div
        class="rounded-lg bg-white shadow-sm p-6  glass bg-gradient-to-br from-white/60 to-white/30 lg:col-span-6"
        style="opacity: 0; transform: translateY(16px);"
      >
        <div
          class="mb-3"
        >
          <h3
            class="text-lg font-semibold"
          >
            Entradas vs saídas (12 meses)
          </h3>
        </div>
        <div
          class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-56 w-full"
        />
        <a
          class="mt-3 inline-flex items-center gap-1 text-sm font-medium text-emerald-700 hover:underline"
          href="/financas/anual"
        >
          Ver detalhes
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right size-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </a>
      </div>
      <div
        class="rounded-lg bg-white shadow-sm p-6  glass bg-gradient-to-br from-white/60 to-white/30 lg:col-span-4"
        style="opacity: 0; transform: translateY(16px);"
      >
        <div
          class="mb-3"
        >
          <h3
            class="text-lg font-semibold"
          >
            Despesas por categoria
          </h3>
        </div>
        <div
          class="rounded-xl border bg-white dark:bg-slate-900 p-4 h-[360px]"
        >
          <div
            class="animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-full w-full"
          />
        </div>
        <a
          class="mt-3 inline-flex items-center gap-1 text-sm font-medium text-emerald-700 hover:underline"
          href="/financas/mensal"
        >
          Ver detalhes
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right size-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </a>
      </div>
      <div
        class="rounded-lg bg-white shadow-sm p-6  glass bg-gradient-to-br from-white/60 to-white/30"
        style="opacity: 0; transform: translateY(16px);"
      >
        <div
          class="mb-3"
        >
          <h3
            class="text-lg font-semibold"
          >
            Contas a vencer
          </h3>
        </div>
        <div
          class="flex flex-col items-center justify-center gap-2 py-10 text-center text-fg-muted"
        >
          <h3
            class="text-sm font-medium"
          >
            Nenhuma conta a vencer
          </h3>
          <div
            class="mt-2"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-8 rounded-md px-3 text-xs"
              style="opacity: 0; transform: translateY(-4px);"
              tabindex="0"
            >
              Nova transação
            </button>
          </div>
        </div>
        <a
          class="mt-3 inline-flex items-center gap-1 text-sm font-medium text-emerald-700 hover:underline"
          href="/financas/mensal"
        >
          Ver detalhes
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-right size-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </a>
      </div>
      <div
        class="rounded-lg bg-white shadow-sm p-6  u-card-base lg:col-span-4"
        style="opacity: 0; transform: translateY(16px);"
      >
        <div
          class="mb-3"
        >
          <h3
            class="text-lg font-semibold"
          >
            Impacto de desejos comprados
          </h3>
        </div>
        <div
          class="flex flex-col items-center justify-center gap-2 py-10 text-center text-fg-muted"
        >
          <h3
            class="text-sm font-medium"
          >
            Sem desejos comprados
          </h3>
        </div>
      </div>
      <div
        class="rounded-lg bg-white shadow-sm p-6  u-card-base lg:col-span-6"
        style="opacity: 0; transform: translateY(16px);"
      >
        <div
          class="mb-3"
        >
          <h3
            class="text-lg font-semibold"
          >
            Lançamentos recentes
          </h3>
        </div>
        <div
          class="flex flex-col items-center justify-center gap-2 py-10 text-center text-fg-muted"
        >
          <h3
            class="text-sm font-medium"
          >
            Nenhuma conta
          </h3>
        </div>
      </div>
      <div
        class="col-span-12 rounded-2xl shadow/soft bg-background/60 backdrop-blur border border-white/10 p-4"
      >
        <h3
          class="font-display tracking-tight text-text dark:text-text-dark text-xl font-semibold mb-3"
        >
          Transações recentes
        </h3>
        <div
          class="w-full animate-shimmer rounded bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-40"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
